lib/declare_schema/model/index_definition.rb

```diff
@@ -33,28 +33,30 @@ def initialize(model, fields, **options)
 
       class << self
         # extract IndexSpecs from an existing table
-        # always includes the PRIMARY KEY index
+        # always includes the PRIMARY KEY index for Model unless it is a HABTM Model.
         def for_model(model, old_table_name = nil)
           t = old_table_name || model.table_name
 
-          primary_key_columns = Array(model.connection.primary_key(t)).presence or
+          habtm_model = model.is_a?(DeclareSchema::Model::HabtmModelShim)
+
+          primary_key_columns = Array(model.connection.primary_key(t)).presence || habtm_model or
```

Comment:
It's a bit crude and uninformative to have the `primary_key_columns` be set to `true` when the `habtm_model` exists. TODO: make this better

Reply:
we can break this into two lines to make it cleaner
```suggestion
          primary_key_columns = Array(model.connection.primary_key(t)).presence 
          primary_key_columns || habtm_model or raise ...
```

Reply:
Nice! That separation definitely _makes this better_